### DISCUSSION TITLE
First Release Checklist

### DISCUSSION THREAD

**Discussion by nvarner** [nvarner/typst-lsp#15] [Announcements]
Thanks to the many PRs we're getting, the LSP is nearly ready for its first release! Here's what has to happen first:

- [x] Decide on a name (depends on [getting a response on Discord about the word "Typst"](https://discord.com/channels/1054443721975922748/1054443722592497796/1088851771553751081))
- [x] Tooling, linting, and formatting configuration
  - [x] Merge TypeScript lints from the `preview` branch (if not covered by #7)
  - [x] Add `prettier` support to TS (#16)
- [x] Add CI support to ensure future PRs compile with high-quality code, deployments happen automatically, etc
  - [x] Merge #12 
  - [x] Merge #7
  - [x] Add CI support for Typescript in the extension
  - [x] Add CI support for running Rust and TS tests (which do not yet exist, but will need to in the near future)
  - [x] Run Clippy on Rust code
- [x] Fix critical bugs
  - [x] Merge #9 
  - [x] Merge #13
 - [x] Prepare for deployment!
   - [x] Get credentials for VS Code Marketplace
   - [x] Add support for Open VSX

If anyone would like to suggest a name, please feel free to do so here.

**Comment by astrale-sharp** [comment]
- Add prettier support to TS
as a means to format typst code?
-  Tooling, linting, and formatting configuration 
I Feel like this is something the community is working on (a package manager was mentionned [here](https://discord.com/channels/1054443721975922748/1088371867913572452/1088836735263313920)

Ideally they might be crates depending on typst as well which would make them quite easy to integrate.

This will take time tho, I would be in favor of publishing without waiting for good tools, after all typst is still in beta.

**Reply by nvarner** [reply]
Sorry, that might have been unclear. I meant to integrate tooling, linting, and formatting for the source code of the LSP.

For example, `prettier` is a formatter for TypeScript. Integrating it here means all the extension source code will have consistent formatting.

**Reply by nvarner** [reply]
Formatting, linting, etc are long-term goals to provide to Typst, but certainly not release blockers :)

**Reply by astrale-sharp** [reply]
great :)

**Comment by beeb** [comment]
For the formatting, I would be partisan of making it both a lib crate and a binary that can be run on any number of input files just like `rustfmt`. The library could then be included in the LSP and the binary could be distributed for people to use in CI etc.

I started looking into making a formatter this afternoon but I have never done something like this and it's a daunting task. I have the SyntaxNode tree but then to enforce rules on this is no easy task for me. In case of a breakthrough I'll make it public.

**Reply by astrale-sharp** [reply]
I start working on it today, feel free to discuss it on the discord with me ;)

**Reply by beeb** [reply]
Cool! I started working on "compile-on-save" vs "compile-on-type"

**Comment by laurmaedje** [comment]
Hey, this is Laurenz from Typst. Thanks for asking about the name and logo!

You are free to use the name "Typst LSP", but only in a non-commercial context. The same holds true for the Typst logo. If you use it, please reproduce it accurately (colors/fonts). You can use the version from our website. Should you decide to create a modified version of the logo, please do it in a way that makes it clearly distinguishable from our logo.

For all commercial usages of our name or logo, you would need to obtain permission from us.

**Reply by nvarner** [reply]
That's great to hear. Thank you!

I have no intention to commercialize the LSP or extension, so I'll stick with the name.

**Reply by laurmaedje** [reply]
> I have no intention to commercialize the LSP or extension, so I'll stick with the name.

I figured as much, just wanted to state this here for the sake of completeness. :)

**Comment by nvarner** [comment]
We've released!

https://open-vsx.org/extension/nvarner/typst-lsp
https://marketplace.visualstudio.com/items?itemName=nvarner.typst-lsp
